packages/arb-ts/src/lib/L1ToL2Message/L1ToL2Message.ts

```diff
@@ -31,86 +29,123 @@ export interface L1ToL2MessageReceipt {
   status: L1ToL2MessageStatus
 }
 
-export class L1ToL2Message extends MultiChainConnector {
-  arbRetryableActions: RetryableActions
-  l1TxnHash?: string
+/**
+ * Conditional type for Signer or Provider. If T is of type Provider
+ * then L1ToL2MessageReaderOrWriter<T> will be of type L1ToL2MessageReader.
+ * If T is of type Signer then L1ToL2MessageReaderOrWriter<T> will be of
+ * type L1ToL2MessageWriter.
+ */
+type L1ToL2MessageReaderOrWriter<T extends Provider | Signer> =
+  T extends Provider ? L1ToL2MessageReader : L1ToL2MessageWriter
 
-  constructor(
-    signersAndProviders: SignersAndProviders,
-    public readonly l2TicketCreationTxnHash: string,
-    public readonly messageNumber?: BigNumber,
-    l1TxnHash?: string
-  ) {
-    super()
-    this.initSignersAndProviders(signersAndProviders)
-    this.arbRetryableActions = new RetryableActions(signersAndProviders)
-    this.l1TxnHash = l1TxnHash
-  }
-  static async initFromL1Txn(
-    signersAndProviders: SignersAndProviders,
-    l1Txn: string | TransactionReceipt,
+export class L1ToL2Message {
+  protected static getMessageOrThrow(
```

Comment:
Unused - should this be used inside of fromL1Receipt / fromL1ReceiptAll?

Reply:
yeah, i was before, but i think it's better just to inline. Removed